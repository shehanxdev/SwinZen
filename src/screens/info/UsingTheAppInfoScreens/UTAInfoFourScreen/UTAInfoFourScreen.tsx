import React from 'react';
import { View } from 'react-native';

import { Text } from '@sz/components';
import { tw } from '@sz/config';
import { TextAlignment, TextVariant } from '@sz/constants';

import { BaseInfoScreen } from '../../components';

export function UTAInfoFourScreen() {
  return (
    <BaseInfoScreen>
      <View style={tw`px-4 py-6`} testID="UTAInfoFourScreenTestID">
        <Text variant={TextVariant.Body2Regular} textAlign={TextAlignment.Auto}>
          UTAInfoFourScreen
        </Text>
      </View>
    </BaseInfoScreen>
  );
}
