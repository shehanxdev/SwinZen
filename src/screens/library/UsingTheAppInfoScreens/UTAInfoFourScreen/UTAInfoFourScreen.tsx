import React from 'react';
import { View } from 'react-native';

import { Text } from '@sz/components';
import { tw } from '@sz/config';
import { Color, TextAlignment, TextVariant } from '@sz/constants';

import { BaseLibraryScreen } from '../../components';

export function UTAInfoFourScreen() {
  return (
    <BaseLibraryScreen>
      <View style={tw`mx-4 mt-8`} testID="UTAInfoFourScreenTestID">
        <Text color={Color.Neutral.White} variant={TextVariant.Body1Regular} textAlign={TextAlignment.Auto}>
          {`Choose either “Down the Line” or “Face on”. Keep in mind that if you choose a video that isn’t shot in the correct perspective the system will not deliver any results. Example: If you select Down the Line, but the video you upload is Face On, there will be no data feedback on that upload.
          \nNext, if you’re left-handed, use the selector to indicate that you are left-handed. If you are right-handed the toggle defaults already.
          \nSelect what type of club you are using.
          \n4a. Auto In App Capture Method - The easiest way to upload a video because it is designed to prevent many shooting errors. Click on the big arrow to start the process.
          \n4b. Manual method - If you choose to upload a video from your phone library you must make sure it is trimmed prior to uploading. It must be trimmed right before backswing and just after. Most phone camera systems have a method of editing and trimming a video. Also the fps camera setting must be on 240fps for FO and 120 or 240 fps for DTL.`}
        </Text>
      </View>
    </BaseLibraryScreen>
  );
}
