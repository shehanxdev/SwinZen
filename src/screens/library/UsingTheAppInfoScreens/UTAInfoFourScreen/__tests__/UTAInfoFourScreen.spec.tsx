import { render } from '@testing-library/react-native';
import * as React from 'react';

import { UTAInfoFourScreen } from '../UTAInfoFourScreen';

describe('UTAInfoFour Screen', () => {
  const testID = 'UTAInfoFourScreenTestID';

  const getRenderedScreen = () => render(<UTAInfoFourScreen />);

  it(`should render UTAInfoFour Screen correctly`, () => {
    const renderer = getRenderedScreen();
    const renderTree = renderer.toJSON();
    expect(renderTree).toMatchSnapshot();
  });

  it('should find the UTAInfoFour Screen via testID', () => {
    const { getByTestId } = getRenderedScreen();
    const foundScreen = getByTestId(testID);
    expect(foundScreen).toBeTruthy();
  });
});
