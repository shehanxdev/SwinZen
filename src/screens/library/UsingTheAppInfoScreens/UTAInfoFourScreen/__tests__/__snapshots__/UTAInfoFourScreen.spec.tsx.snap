// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UTAInfoFour Screen should render UTAInfoFour Screen correctly 1`] = `
<BVLinearGradient
  colors={
    [
      4279917603,
      4278984207,
      4278650631,
    ]
  }
  endPoint={
    {
      "x": 0.5,
      "y": 1,
    }
  }
  locations={null}
  startPoint={
    {
      "x": 0.5,
      "y": 0,
    }
  }
  style={
    {
      "flexBasis": "0%",
      "flexGrow": 1,
      "flexShrink": 1,
      "position": "relative",
    }
  }
>
  <Image
    resizeMode="cover"
    source={
      {
        "testUri": "../../../src/assets/images/grassBackground.jpg",
      }
    }
    style={
      {
        "opacity": 0.1,
        "position": "absolute",
      }
    }
  />
  <RCTSafeAreaView
    style={
      {
        "height": "100%",
        "paddingTop": 50,
      }
    }
  >
    <RCTScrollView
      contentContainerStyle={
        {
          "flexGrow": 1,
        }
      }
      keyboardShouldPersistTaps="handled"
    >
      <View>
        <View
          style={
            {
              "marginLeft": 16,
              "marginRight": 16,
              "marginTop": 32,
            }
          }
          testID="UTAInfoFourScreenTestID"
        >
          <Text
            style={
              [
                {
                  "color": "#FFFFFF",
                  "lineHeight": 28,
                },
                {
                  "fontFamily": "SourceSansPro-Regular",
                  "fontSize": 18,
                  "fontWeight": "normal",
                  "textAlign": "auto",
                },
              ]
            }
          >
            Choose either “Down the Line” or “Face on”. Keep in mind that if you choose a video that isn’t shot in the correct perspective the system will not deliver any results. Example: If you select Down the Line, but the video you upload is Face On, there will be no data feedback on that upload.
          
Next, if you’re left-handed, use the selector to indicate that you are left-handed. If you are right-handed the toggle defaults already.
          
Select what type of club you are using.
          
4a. Auto In App Capture Method - The easiest way to upload a video because it is designed to prevent many shooting errors. Click on the big arrow to start the process.
          
4b. Manual method - If you choose to upload a video from your phone library you must make sure it is trimmed prior to uploading. It must be trimmed right before backswing and just after. Most phone camera systems have a method of editing and trimming a video. Also the fps camera setting must be on 240fps for FO and 120 or 240 fps for DTL.
          </Text>
        </View>
      </View>
    </RCTScrollView>
  </RCTSafeAreaView>
</BVLinearGradient>
`;
